lib/active_storage_validations/analyzer/image_analyzer/image_magick.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module ActiveStorageValidations
+  # This analyzer relies on the third-party {MiniMagick}[https://github.com/minimagick/minimagick] gem.
+  # MiniMagick requires the {ImageMagick}[http://www.imagemagick.org] system library.
+  # This is the default Rails image analyzer.
+  class Analyzer::ImageAnalyzer::ImageMagick < Analyzer::ImageAnalyzer
+
+    private
+
+    def read_image
+      begin
+        require "mini_magick"
```

Comment:
question if we need to require it always here
can we just check if defined?(MiniMagick)?

and maybe raise the error even sooner, in the initializer (check when the app boots if possible)?

Reply:
I think we do not need to require it every time. That's, indeed, not necessary, plus it has performance benefits.

I am updating the code with a `unless defined?(MiniMagick)` (same for Vips), as it was done previouslu by the `metadata.rb` file (I should have ketp it).

For the second part, we could raise the error at the app boot, but I think this will enforce the user to have both `imagemagick` and `ruby-vips` gem in their `Gemfile` (which we do not want). Plus, if the user does not use validators based on the image analyzers, it's not necessary to have either `imagemagick` or `ruby-vips` gem. Therefore I think it's better to leave it this way? What do you think?

Reply:
Using `Object.const_defined?(:Vips)` is better than `unless defined?(Vips)` because we might not have loaded the lib yet, therefore it might not be defined yet. Whereas `Object.const_defined?(:Vips)` is about presence in the code.

Reply:
I'm fine with bject.const_defined?(:Vips)

I'm also fine if it just raises an exception and we will have something about it in the docs. Not the most user friendly but this should be expected. Bu this option is least preferable.

Reply:
@igorkasyanchuk I finally opted for `defined?(MiniMagick)`. It is the best option you are right. Let me know if the PR is okay with you now :)